src/spikeinterface/sortingcomponents/tools.py

```diff
@@ -151,3 +300,21 @@ def fit_sigmoid(xdata, ydata, p0=None):
 
     popt, pcov = curve_fit(sigmoid, xdata, ydata, p0)
     return popt
+
+
+def get_shuffled_recording_slices(recording, seed=None, **job_kwargs):
+    from spikeinterface.core.job_tools import ensure_chunk_size
+    from spikeinterface.core.job_tools import divide_segment_into_chunks
+
+    chunk_size = ensure_chunk_size(recording, **job_kwargs)
+    recording_slices = []
+    for segment_index in range(recording.get_num_segments()):
+        num_frames = recording.get_num_samples(segment_index)
+        chunks = divide_segment_into_chunks(num_frames, chunk_size)
+        recording_slices.extend([(segment_index, frame_start, frame_stop) for frame_start, frame_stop in chunks])
+
+    if seed is not None:
+        rng = np.random.RandomState(seed)
+        recording_slices = rng.permutation(recording_slices)
```

Comment:
```suggestion

    recording_slices = rng.permutation(recording_slices)
```

Reply:
this will not work if rng is not created, without seed, will it?